sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/Utilities/UtilityAnalyzerBase.cs

```diff
@@ -154,20 +155,7 @@ protected sealed override void Initialize(SonarAnalysisContext context)
                         // Make sure the folder exists
                         Directory.CreateDirectory(WorkDirectoryBasePath);
 
-                        if (!File.Exists(pathToWrite))
-                        {
-                            FileStream fileStream = null;
-                            try
-                            {
-                                fileStream = File.Create(pathToWrite);
-                            }
-                            finally
-                            {
-                                fileStream?.Dispose();
-                            }
-                        }
-
-                        using (var metricsStream = new FileStream(pathToWrite, FileMode.Append, FileAccess.Write))
+                        using (var metricsStream = File.Create(pathToWrite))

```

Comment:
We now always overwrite existing files, this was mainly made for testing, but the old behavior does not seem correct. We are creating a new folders for the protobuf files on each build, so we were always creating a new file.

Reply:
Could have maybe be part of another commit.